<launch>
    <catch2_launch_file description='An example Catch2 integration test launch file'/>

    <node pkg='nuturtle_control' exec='odometry_node'>
        <param name="body_id" value="base_footprint" />
        <param name="odom_id" value="odom" />
        <param name="wheel_left" value="wheel_left" />
        <param name="wheel_right" value="wheel_right" />
        <param name="wheel_radius" value="0.033" />
        <param name="track_width" value="0.160" />
    </node>

    <!-- Launch joint_state_publisher -->
    <node pkg="joint_state_publisher" exec="joint_state_publisher">
        <param name="use_gui" value="false"/>
    </node>

    <catch2_integration_test_node
        pkg='nuturtle_control'
        exec='odometry_test_node'
    >

    </catch2_integration_test_node>
</launch>